src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionImportControlTest.java

```diff
@@ -120,7 +120,7 @@ public void testFour() throws Exception {
         };
 
         final List<String> expectedXpathQueries = Collections.singletonList(
-            "/IMPORT[./DOT[@text='Scanner']]"
+            "/IMPORT[2]"
```

Comment:
@timurt , please explain this update. Original version were more strict. updated version will match to everything.

Reply:
@romani 

The point is - previous implementation of the `XpathQueryGenerator` iterated through node's children and checked if child node has `IDENT` child node.
The problem is
1. It is possible that node with `@text` attribute will be somewhere in the subtree (may be grand grand child of the current node), so we need to check recursively all subtree, it could take a lot of time resources
2. Finding node with `@text` attribute does not guarantee that node is unique among its siblings - simple example is `/IMPORT[./DOT[@text='Scanner']]` can be applied for both `java.util.Scanner` and `kz.tim.Scanner`

That's why I think `/IMPORT[2]` is more strict than `/IMPORT[./DOT[@text='Scanner']]`, because you can have several `Scanner` imports

New implementation of the `XpathQueryGenerator` works in following way
if our node 
1. does not support `@text` attribute
and
2. is not unique among siblings
and 
3. has no child which supports `@text` attribute
then simply find position among siblings

If you think we should make xpath query more strict then I can
1. additionally check grandchildren which support `@text` attribute
or
2. recursively iterate subtree until I find node with `@text` attribute

Reply:
can we have xpath, like https://stackoverflow.com/a/5818966/1015848 ?

Reply:
You think it should be something like this `(/IMPORT[./DOT/IDENT[@text='Scanner']])[1]`?

Reply:
Yes, expression with some text from file it better, than just index.

Reply:
done